### PR TITLE
RPC: `descriptorprocesspsbt` returns hex encoded tx if complete

### PR DESCRIPTION
**Pull request by ismaelsadeeq**
Coming from [#28414 comment](https://github.com/bitcoin/bitcoin/pull/28414#pullrequestreview-1618684391) Same thing also for `descriptorprocesspsbt`.

Before this PR `descriptorprocesspsbt` returns a boolean `complete` which indicates that the psbt is final, users then have to call `finalizepsbt` to get the hex encoded network transaction.

In this PR if the psbt is complete the return object also has the hex encoded network transaction ready for broadcast with `sendrawtransaction`.

This save users calling `finalizepsbt` with the descriptor, if it is already complete.

### CODE REVIEW

File: test/functional/rpc_psbt.py [hunk 1]
```diff
@@ -978,17 +978,22 @@ def test_psbt_input_keys(psbt_input, keys):
...
+        decoded = self.nodes[2].decodepsbt(processed_psbt['psbt'])
         test_psbt_input_keys(decoded['inputs'][0], ['witness_utxo', 'non_witness_utxo', 'partial_signatures', 'bip32_derivs'])
 
-        psbt = self.nodes[2].descriptorprocesspsbt(psbt=psbt, descriptors=[descriptor], sighashtype="ALL", bip32derivs=False, finalize=True)["psbt"]
-        decoded = self.nodes[2].decodepsbt(psbt)
+        # If psbt not finalized, test that result does not have hex
+        assert "hex" not in processed_psbt
+
+        processed_psbt = self.nodes[2].descriptorprocesspsbt(psbt=psbt, descriptors=[descriptor], sighashtype="ALL", bip32derivs=False, finalize=True)
+        decoded = self.nodes[2].decodepsbt(processed_psbt['psbt'])
```
Comment (pinheadmz) [inline test/functional/rpc_psbt.py:989]: nit: might be cleaner to not re-use variable names `processed_psbt` and `decoded` but im not sure how strict we are about stuff like that in tests

**Review by pinheadmz — APPROVED** [review]
ACK a99e9e655a58b2364a74aec5cafb827a73c6b0c4

reviewed code and tested locally :+1:

<details><summary>Show Signature</summary>

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

ACK a99e9e655a58b2364a74aec5cafb827a73c6b0c4
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEE5hdzzW4BBA4vG9eM5+KYS2KJyToFAmUEszgACgkQ5+KYS2KJ
yTqdKA//XiTF0aXFKBoFAxJLU1oPrMNTD5RpgP9mvSu0rHuKf8g1KvAqUv1doeUH
A2/RpbWPxHJyivbg7S3h3Dfzfrzbh5TRCO+47zSiIKzINtJ92VG1CXwe+Uu7O35v
Ez6eUBk6tYeZOo+udIbtKXhQFfE8MQ8IiJ/qjJj4UGw1ehvJZ9FiKF80JHGv7HK0
pRVrZqNsYpQ44DYuwAxUJQuiz0uX/hPulpheAxdk1kAoFaVmWXH+YEhXfuvqwIPc
FyTZE/BWaVYTQaT9S0bT0QXntq5L9x/jQI9fANrPZ10REpySHgD9uuJxfF1ZZ077
rlLgq7S9+kY8kTr+mfXSgBJdr/dHQRnY9yrFrwUbLOZx0DI6Y439k0V1ZXmbuLmb
xVQ78rNNomEUE3KznhiTXAmFiJEYKOnpEqOf8Dny/UQ4nuNUvEtzPJmm2f4hFkfC
xM6C9bjZp2oYFyE0dKpp2Pcw6TDroUW8kdt4R21hm4rwWZ2gz0/JHlimPQP1qtPf
PFy5gWSMM9P0z4F5xFZfqgaPDCxYEPQ5Zsu/WonS1oE8glIhVPLeE++vAtd0iqAZ
Ibs8qTFQDaZUlfkdyEVB+Bx+OykD67sFUd5f62579d08LeCaat1Hlddk0tuWbNXs
TcTop9JnD9ihmWGyU6oM8agHO8EqXmJDccfEWXxf52BkutECzhM=
=R4/T
-----END PGP SIGNATURE-----
```

pinheadmz's public key is [on keybase](https://keybase.io/pinheadmz/pgp_keys.asc?fingerprint=e61773cd6e01040e2f1bd78ce7e2984b6289c93a)
</details>

**Review by ishaanam — COMMENTED** [review]
ACK a99e9e655a58b2364a74aec5cafb827a73c6b0c4

File: doc/release-notes-28414.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 RPC Wallet
 ----------
 
-- RPC `walletprocesspsbt` return object now includes field `hex` (if the transaction
-is complete) containing the serialized transaction suitable for RPC `sendrawtransaction`. (#28414)
+- RPC `walletprocesspsbt`, and `descriptorprocesspsbt` return object now includes field `hex` (if the transaction
+is complete) containing the serialized transaction suitable for RPC `sendrawtransaction`.
```
[the remark this answers is not in the document]
Comment (ismaelsadeeq) [reply doc/release-notes-28414.md:5]: Thanks, this basically means the same thing IMO,  resolving this.

File: test/functional/rpc_psbt.py (hunk 1 above)
Comment (ismaelsadeeq) [reply test/functional/rpc_psbt.py:989]: Thanks for your review!
Will use distinct variable names in future contribution, but happy to update this if another reviewer ask for it.

**Comment by achow101** [comment]
ACK a99e9e655a58b2364a74aec5cafb827a73c6b0c4

**Pull request merged**